weatherbenchX/metrics/categorical.py

```diff
@@ -487,3 +487,717 @@ def _values_from_mean_statistics_with_internal_names(
+    assert self._threshold_probability_dim in statistic_values['TruePositives'].dims, (
+        f"Threshold probability dim {self._threshold_probability_dim} not found in "
+        f"TruePositives statistic"
+    )
+    hit_rate = statistic_values['TruePositives'] / (
+        statistic_values['TruePositives'] + statistic_values['FalseNegatives']
+    )
+    # print(hit_rate)
+    false_alarm_rate = statistic_values['FalsePositives'] / (
+        statistic_values['FalsePositives'] + statistic_values['TrueNegatives']
+    )
+    # print(false_alarm_rate)
+    auc_dims = tuple(set(hit_rate.dims) - {self._threshold_probability_dim})
+
+    auc = -1 * xr.apply_ufunc(
+      np.trapezoid,
+      hit_rate,
+      false_alarm_rate,
+      input_core_dims=[hit_rate.dims, false_alarm_rate.dims],  # type: ignore
+      output_core_dims=[auc_dims],
+      dask="allowed",
+    )
+
+    return 2 * auc - 1
+
+
+class REV(base.PerVariableMetric):
+  """ Relative economic value (REV)
+
+  Definition in:
+  Richardson, D. S. (2000). Skill and relative economic value of the ECMWF ensemble prediction system. 
+      Quarterly Journal of the Royal Meteorological Society, 126(563), 649–667. https://doi.org/10.1002/qj.49712656313
+  Richardson, David S. (2006). Predictability and economic value. In T. Palmer & R. Hagedorn (Eds.), 
+      Predictability of Weather and Climate (1st ed., pp. 628–644). Cambridge University Press. https://doi.org/10.1017/CBO9780511617652.026
+  Wilks, D. S. (2001). A skill score based on economic value for probability forecasts. 
+      Meteorological Applications, 8(2), 209–219. https://doi.org/10.1017/S1350482701002092
+
+  total = TP + FP + FN + TN
+  e_clim = min{(TP + FN) / total, cost_loss_ratios}
+  e_fcst = (TP + FP) / total * cost_loss_ratios + FN / total
+  e_perfect = (TP + FN) / total * cost_loss_ratios
+
+  REV = (e_clim - e_fcst) / (e_clim - e_perfect)
+  """
+  def __init__(
+      self,
+      cost_loss_ratios: Union[float, Sequence[float]],
+  ):
+    """Init.
+
+    Args:
+      cost_loss_ratios: Cost loss ratios for calculating the REV.
+    """
+    self._cost_loss_ratios = xr.DataArray(
+        cost_loss_ratios, dims=["cost_loss_ratios"], 
+        coords={"cost_loss_ratios": cost_loss_ratios}
+    )
+  
+
+  @property
+  def statistics(self) -> Mapping[Hashable, base.Statistic]:
+    return {
+        'TruePositives': TruePositives(),
+        'FalsePositives': FalsePositives(),
+        'FalseNegatives': FalseNegatives(),
+        'TrueNegatives': TrueNegatives(),
+    }
+
+  def _values_from_mean_statistics_per_variable(
+      self,
+      statistic_values: Mapping[Hashable, xr.DataArray],
+  ) -> xr.DataArray:
+    """Computes metrics from aggregated statistics."""
+
+    event_count = statistic_values['TruePositives'] + statistic_values['FalseNegatives']
+    event_nonevent_count = event_count + statistic_values['FalsePositives'] + statistic_values['TrueNegatives']
```

Comment:
FYI this should always be 1 (up to numerical error) because weatherbench gives you *means* of the statistics here. The entries of the confusion matrix always sum to 1 on a per-example basis, and so their means should sum to 1 as well.

In the case of numerical error I'm actually not 100% sure what's best to do here, but I don't think it would generally make a big difference so perhaps better to go with the simpler option and avoid the extra division?